doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,812 @@
+
+        /// Gets the app catalog  where this package version is from.
+        AppCatalog AppCatalog { get; };
+
+        /// DESIGN NOTE:
+        /// GetManifest from IPackageVersion in AppInstallerRepositorySearch is not implemented in V1. That class has a lot of fields and no one requesting it.
+        /// Gets the manifest of this package version.
+        /// virtual Manifest::Manifest GetManifest() = 0;
+    };
+
+    /// IMPLEMENTATION NOTE: PackageVersionKey from AppInstallerRepositorySearch.h
+    /// A key to identify a package version within a package.
+    [version(1)]
+    struct PackageVersionId
+    {
+        /// The app catalog id that this version came from.
+        String AppCatalogId;
+        /// The version.
+        String Version;
+        /// The channel.
+        String Channel;
+    };
+
+    /// IMPLEMENTATION NOTE: IPackage from AppInstallerRepositorySearch.h
+    /// A package, potentially containing information about it's local state and the available versions.
+    runtimeclass Package
+    {
+        Package();
+
+        /// IMPLEMENTATION NOTE: This would be new work as Windows Package Manager does not currently create a Package object from a manifest.
+        /// DESIGN NOTE: This is proposed as the preferred alternative to having InstallOptions match the winget command line which
+        /// allows you to pass in a manifestPath directly into install. Allowing creation of a Package object from a manifest allows the install api to 
+        /// consistently use package objects rather than having a "you can either set the manifest path or use a package but not both" problem.
+        Package(String manifestPath);
+
+        /// IMPLEMENTATION NOTE: PackageProperty fields from AppInstallerRepositorySearch.h
+        /// Gets a property of this package.
+        String Id { get; };
+        String Name { get; };
+
+        /// Gets the installed package information if the package is installed.
+        PackageVersion InstalledVersion{ get; };
+
+        /// Gets all available versions of this package.
+        /// The versions will be returned in sorted, descending order.
+        ///  Ex. { 4, 3, 2, 1 }
+        Windows.Foundation.Collections.IVectorView<PackageVersionId> AvailableVersions { get; };
+
+        /// Gets a specific version of this package.
+        PackageVersion LatestAvailableVersion { get; };
+
+        /// Gets a specific version of this package.
+        PackageVersion GetAvailableVersion(PackageVersionId versionKey);
+
+        /// Gets a value indicating whether an available version is newer than the installed version.
+        Boolean IsUpdateAvailable { get; };
+
+        /// Gets whether the package is installing
+        Boolean IsInstalling { get; };
+        /// Gets the installing progress if the package is installing
+        Windows.Foundation.IAsyncOperationWithProgress<InstallResult, InstallProgress>  InstallProgress{ get; };

```

Comment:
I know we talked about this at length in the `PackageCatalog` time. I think we settled on this not being an event on the Package object (since the Package object is just "data") and instead "this is being installed" goes on the catalog (or catalog manager) instead.

Reply:
I think IsInstalling still counts as CatalogPackage data? Or at least, it seems like it fits with the rest of the design here where the CatalogPackage object contains a bunch of data about the installed version of the package. Removing the Installing and Installed data from Package would defeat the purpose of GetCompositeAppCatalog since that feature is (mostly) designed to give you one Package object that has data from multiple sources. I have moved the InstallProgress property to be a method next to the InstallPackageAsync method now though.